client/src/main/java/io/pravega/client/batch/impl/SegmentIteratorImpl.java

```diff
@@ -43,14 +50,19 @@ public class SegmentIteratorImpl<T> implements SegmentIterator<T> {
     private final long startingOffset;
     private final long endingOffset;
     private final EventSegmentReader input;
+    private final long timeoutInMilli = 2000L;
```

Comment:
This seems needlessly aggressive given that there are no retries or way to configure the value.

Reply:
Can we use clientConfig defaultTimeout here?

Reply:
Updated to use timeout from clientConfig.